src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseGradingService.java

```diff
@@ -740,9 +753,12 @@ private double calculatePointsForSuccessfulTestCase(final Result result, final P
         final boolean isWeightSumZero = Precision.equals(weightSum, 0, 1E-8);
         final double testPoints;
 
-        // a non-zero score has to be calculated for the solution even if the weight sum is zero
-        // to avoid a warning to the instructor that the solution has faults
-        // => each test case receives a weight of one
+        // If the weight sum is zero, then the solution would receive zero points as well even if all tests pass, thus
+        // resulting in a warning to the instructor. The warning is wrong however, as setting a weight sum of zero is
+        // only possible for exercises with manual feedback, i.e., if all points should only be explicitly be awarded
+        // manually by tutors. This is obviously not possible for the solution.
+        // Giving each test case a weight of one makes sure that a score > 0% can be reached by the solution and the
+        // 100% needed to not generate a warning is only possible if all tests pass.
         if (isWeightSumZero && result.getParticipation() instanceof SolutionProgrammingExerciseParticipation) {
```

Comment:
Thanks for giving a more thorough explanation. Now I know what you mean. I'd suggest the following description to not publish a book aside with the source code :D

> A weight-sum of zero would let the solution show an error to the instructor as the solution score must be 100% of all reachable points.
To prevent this, we weigh all test cases equally in such a case.

Reply:
:+1: